apps/rmh-web/components/SideBar.tsx

```diff
@@ -0,0 +1,76 @@
+"use client";
+import { User } from "@/app/pt-portal/models";
+import Image from "next/image";
+import { useMemo, useState } from "react";
+
+import UsersList from "./UsersList";
+import Tab from "./Tab";
+
+const tabs = ["Today's", "All"];
+
+const useFinalUsers = (activeIndex: number, search: string, users: User[]) =>
+  useMemo(() => {
+    const userFilter = (user: User) =>
+      new Date(user.appointmentDate).getTime() > new Date().getTime();
+    const filteredUsers = activeIndex === 0 ? users.filter(userFilter) : users;
+
+    return search.trim().length > 0
+      ? filteredUsers.filter((user) =>
+          user.patientName.toLowerCase().startsWith(search)
+        )
+      : filteredUsers;
+  }, [activeIndex, search, users]);
+
+const SideBar = ({
+  isLoading,
+  users,
+}: {
+  isLoading: boolean;
+  users: User[];
+}) => {
+  const [search, setSearch] = useState("");
```

Comment:
this function should probably do validation and only allow [a-Z] type stuff.  Prevent typing that starts with an empty space as well